engine/src/main/java/org/apache/apex/engine/plugin/loaders/StaticPluginLocator.java

```diff
@@ -19,31 +19,34 @@
 package org.apache.apex.engine.plugin.loaders;
 
 import java.util.Arrays;
-import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import org.apache.apex.api.plugin.Plugin;
 import org.apache.apex.engine.api.plugin.PluginLocator;
 import org.apache.hadoop.conf.Configuration;
 
 /**
  * @since 3.6.0
  */
-public class StaticPluginLocator<T> implements PluginLocator<T>
+public class StaticPluginLocator<T extends Plugin> implements PluginLocator<T>
 {
   private static final Logger LOG = LoggerFactory.getLogger(StaticPluginLocator.class);
 
-  private final T[] plugins;
+  private final Set<T> plugins;
 
   public StaticPluginLocator(T... plugins)
   {
-    this.plugins = plugins;
+    this.plugins = Collections.unmodifiableSet(new HashSet(Arrays.asList(plugins)));
```

Comment:
Sets.newHashSet() calls `new HashSet()`. Its usage was justifiable prior to java 1.7. There is no need to use it now unless it can help to construct unmodifiable Set from an array directly.